### ISSUE TITLE
torch.cat produces wrong results when concat dim has size 1

### ISSUE THREAD

**Issue by raghukiran1224** [torch-spyre/torch-spyre#1760]
## Summary

`torch.cat([cache, new], dim=2)` produces incorrect results on Spyre when `new` has size 1 along the concatenation dimension. The same operation at size 64 is accurate (max diff 0.002).

This is the root cause of decode-step numerical drift in autoregressive LLM generation on Spyre — every decode step concatenates a size-1 K/V tensor into the KV cache.

## Minimal Reproducer

```python
"""torch.cat bug on Spyre: size-1 concat dimension produces wrong results.

PYTHONPATH=. python3 test_cat_minimal.py
"""
import torch
import torch_spyre  # noqa: F401

torch.manual_seed(42)

cache = torch.randn(1, 8, 64, 128, dtype=torch.float16)

for new_seq in [64, 1]:
    new = torch.randn(1, 8, new_seq, 128, dtype=torch.float16)
    cpu_out = torch.cat([cache, new], dim=2)

    def cat_fn(c, n):
        return torch.cat([c, n], dim=2)

    compiled = torch.compile(cat_fn, dynamic=False)
    spyre_out = compiled(cache.to("spyre"), new.to("spyre"))

    diff = (cpu_out.float() - spyre_out.to("cpu").float()).abs()
    print(f"new_seq={new_seq}: max_diff={diff.max().item():.4f}  mean_diff={diff.mean().item():.6f}")

# Expected:
#   new_seq=64: max_diff=0.0020  (correct)
#   new_seq=1:  max_diff=4.5664  (broken)
```

## Results

```
new_seq=64: max_diff=0.0020  mean_diff=0.000280   ← correct
new_seq=1:  max_diff=4.0967  mean_diff=0.025956   ← broken
```

Tested on torch-spyre @ 7c6ef99.

## Impact

This breaks autoregressive LLM decode on Spyre. The KV cache update in every decoder layer does `torch.cat([key_cache, new_key], dim=2)` where `new_key` has `seq_len=1` during decode. The corrupted K/V values cause SDPA to produce wrong attention outputs, which compound across layers and tokens.

Prefill (seq_len=64) is unaffected — the first generated token is correct. Only subsequent decode tokens drift.

## Bisection Details

Isolated via progressive fusion testing of a transformer decoder block:

| Sub-block | seq=64 max diff | seq=1 max diff |
|-----------|----------------|----------------|
| RMSNorm alone | 0.02 | 0.02 |
| Linear + reshape alone | 0.009 | 0.008 |
| `torch.cat` alone | **0.002** | **4.57** |
| Linear + reshape + cat | 0.009 | 4.19 |
| Full block (norm→QKV→RoPE→cat→SDPA→MLP) | 0.008 | 0.19 (SDPA averages out the cat error) |

The error is in `torch.cat` itself, not in op fusion — Test 1 above shows it with `torch.cat` compiled standalone.

**Comment by raghukiran1224** [comment]
Additional data — the error is a sharp cliff at exactly size 1, not a gradual degradation:

```
new_seq=64: max_diff=0.0020   OK
new_seq=32: max_diff=0.0020   OK
new_seq=16: max_diff=0.0020   OK
new_seq= 8: max_diff=0.0020   OK
new_seq= 4: max_diff=0.0020   OK
new_seq= 2: max_diff=0.0020   OK
new_seq= 1: max_diff=4.9248   BROKEN
```

Sizes 2–64 all produce identical 0.002 max diff. Only size 1 is wrong. This suggests a special-case code path for size-1 dimensions in the stickify/layout logic.

**Comment by raghukiran1224** [comment]
Related: #1719 (torch.cat numerical mismatch in eager mode across multiple shapes/dtypes).

This issue adds:
- A standalone minimal reproducer (no external test framework)
- Bisection showing the bug is size-dependent: size-1 concat dim fails, size-64 is correct
- Compiled path coverage (torch.compile), whereas #1719 covers eager mode
- The connection to autoregressive decode drift — prefill works, decode breaks because every KV cache update concats a size-1 tensor

**Comment by avery-blanchard** [comment]
Hi @raghukiran1224 ! 
We recently merged fixes for this. Could you check which commit you are on?
```
new_seq=64: max_diff=0.0020  mean_diff=0.000280
new_seq=1: max_diff=0.0020  mean_diff=0.000280
```

**Comment by raghukiran1224** [comment]
confirmed that this issue is fixed.. thanks @avery-blanchard
